### ISSUE TITLE
objects with all optional properties should not be able to contain other properties

### ISSUE THREAD

**Issue by k4b7** [microsoft/TypeScript#1327]
```
interface Options {
   foo?: boolean;
   bar?: boolean;
}

var options: Options;
options = { foobar: true };  // should raise a compiler error but doesn't
```

As soon as one of the options is not optional then we get the expected behaviour.

```
interface Options {
   foo: boolean;
   bar?: boolean;
}

var options: Options;
options = { foobar: true };  // raises a compiler error
```
